src/npc.cpp

```diff
@@ -1700,6 +1701,12 @@ void npc::shop_restock()
         }
     }
 
+    // This removes some items according to item spawn scaling factor,
+    const float spawn_rate = get_option<float>( "ITEM_SPAWNRATE" );
+    ret.remove_if( [spawn_rate]( auto &/*i*/ ) {
```

Comment:
```suggestion
    ret.remove_if( [spawn_rate]( auto & ) {
```
You can also just do this.  Depends whether you think that comment is helpful.